apps/i2c-accelerometer-gyroscope/src/i2c-accelerometer-gyroscope.c

```diff
@@ -0,0 +1,150 @@
+
+/*
+ * Note that measured data has 16 bits, what makes necessary to put together
+ * the MSB's with their LSB's.
+ * Example:
+ * buf[0] will have MSB's for x axis acceleration and buf[1] will have LSB's
+ * for x axis acceleration, that is, the raw data of x axis acceleration will
+ * be buf16[0] = (buf[0] << 8) | buf(1)
+ */
+#define MERGE_MSB_LSB(a,b) ( a << 8 | b )
+
+/*
+ * Offset values for sensor calibration
+ * To obtain these values, put the sensor in your chosen origin and realise a
+ * group of measurements, then take the mean value for each variable.
+ * Obs: Note that if you're using a high precision sensibility, the measures
+ * will vary slightly everytime.
+ */
+
+#define ACCEL_X_OFFSET		236
+#define ACCEL_Y_OFFSET		-752
+#define ACCEL_Z_OFFSET		20996
+#define GYRO_X_OFFSET		75
+#define GYRO_Y_OFFSET		-147
+#define GYRO_Z_OFFSET		23
+
+/* Buffer vector to store data from sensor 8 bit registers */
+u8_t register_buffer[14];
+s16_t measured_data[7]; /* Buffer vector to store measured data from sensor */
+
+s32_t accel[3], gyro[3]; /* Each vector position corresponds to an axis, e.g.
+accel[0] -> acceleration in x axis */
+
+LOG_MODULE_REGISTER(accelgyro, LOG_LEVEL_DBG);
+
+struct device *i2c_dev;
+
+void setup(void)
+{
+	/* Peripherals control */
+	i2c_dev = device_get_binding(DT_NORDIC_NRF_I2C_0_LABEL);
+	/* Checks if peripheral was properly configured */
+	if (!i2c_dev) {
+		printk("I2C: Device driver not found.\n");
+
+		return;
+	}
+	/* Take sensor out of sleep mode */
+	i2c_reg_update_byte(i2c_dev, MPU_ADDRESS,
+				MPU6050_REG_PWR_MGMT1, MPU6050_SLEEP_EN, 0);
+	/* Set the full scale range for gyroscope */
+	i2c_reg_write_byte(i2c_dev, MPU_ADDRESS, MPU6050_REG_GYRO_CFG, FS_SEL);
+	/* Set the full scale range for accelerometer */
+	i2c_reg_write_byte(i2c_dev, MPU_ADDRESS,
+				MPU6050_REG_ACCEL_CFG, AFS_SEL);
+
+	char *name_list[6] = { "ACCELX", "ACCELY", "ACCELZ",
+				"GYROX", "GYROY", "GYROZ" };
+	/* KNoT config */
+	for (int i = 0; i<3; i++) {
+		knot_data_register(i, name_list[i], KNOT_TYPE_ID_ACCELERATION
+					KNOT_VALUE_TYPE_INT,
+					KNOT_UNIT_ACCELERATION_MS2,
+					&accel[i], sizeof(accel[i]),
+					NULL, NULL);
+		knot_data_config(i, KNOT_EVT_FLAG_CHANGE, NULL);
+	}
+	for (int k = 0; k<3; k++) {
+		knot_data_register(k+3, name_list[k+3],
+				KNOT_TYPE_ID_ANGULAR_VELOCITY,
+				KNOT_VALUE_TYPE_FLOAT,
+				KNOT_UNIT_ANGULAR_VELOCITY_DEGSEG,
+				&gyro[k], sizeof(gyro[k]), NULL, NULL);
+	}
+
+}
+
+void loop(void)
+{
+	static u64_t last_toggle_time = 0;
+	static int i = 0; /* Buffer vector index */
+
+	/* Merge MSB with LSB */
+	buf16[i] = MERGE_MSB_LSB(register_buffer[2 * i],
+					register_buffer[2 * i + 1]);
+
+	if (i > 6) {
+		i = 0;
+	}
+
+	/* Get current time */
+	int64_t current_time = k_uptime_get();
+	/* Get new measures in a specified time interval */
+	if (current_time - last_toggle_time > TIME_INTERVAL){
+		i2c_burst_read(i2c_dev, MPU_ADDRESS,
+					 MPU6050_REG_DATA_START, buf, 14);
```

Comment:
What is the '14' value? The size of buf? Add a define for that. Avoid using magic numbers.

Reply:
Done